vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -1740,3 +1741,565 @@ def execute_model(
             prompt_logprobs_dict={},
             pooler_output=pooler_output)
         return model_output
+
+
+class ChunkedPrefillModelRunner(ContinuousBatchingSpyreModelRunner):
+
+    def _prepare_prompt(self, _):
+        AssertionError(
+            "Should not call this method on chunked prefill implementation")
+
+    def _prepare_chunked_prefill(self, scheduler_output: SchedulerOutput):
+        '''
+        Cases / Scenarios for the chunked prefill with right padding.
+ 
+
+        X    - Padding
+        T    - Token
+        O    - Right padding / unused slot
+        |    - Blocks separator
+        ||   - Chunks separator
+        ...  - Trimmed sequence
+
+        For the following "drawing" consider blocks of 4 tokens
+        and chunks of 8 tokens
+
+        # Case I
+
+        Prompt fits in the chunk with left padding
+
+        3 tokens
+        1 block
+        1 chunk
+        4 left padding
+
+        X X X X | T T T O || O O O O | O O O O || O O O O | O O O O || ...
```

Comment:
Nice drawings!

I would probably omit the trailing `0 0 0 0` blocks to remove any confusion that we're pre-allocating a long sequence of right padding